src/torchcodec/transforms/_decoder_transforms.py

```diff
@@ -66,28 +72,167 @@ class Resize(DecoderTransform):
+            #       API guarantee with our users.
+            self._top = int(
+                torch.randint(0, self._input_dims[0] - self.size[0] + 1, size=()).item()
+            )
+            self._left = int(
+                torch.randint(0, self._input_dims[1] - self.size[1] + 1, size=()).item()
             )
-        return cls(size=resize_tv.size)
+
+        return f"crop, {self.size[0]}, {self.size[1]}, {self._left}, {self._top}"
+
+    def _get_output_dims(
+        self, input_dims: Tuple[Optional[int], Optional[int]]
+    ) -> Tuple[Optional[int], Optional[int]]:
+        # TODO: establish this invariant in the constructor during refactor
+        assert len(self.size) == 2
+
+        height, width = input_dims
+        if height is None:
+            raise ValueError(
+                "Video metadata has no height. RandomCrop can only be used when input frame dimensions are known."
+            )
+        if width is None:
+            raise ValueError(
+                "Video metadata has no width. RandomCrop can only be used when input frame dimensions are known."
+            )
+
+        self._input_dims = (height, width)
+        return (self.size[0], self.size[1])
+
+    @classmethod
+    def _from_torchvision(
+        cls,
+        tv_random_crop: nn.Module,
+        input_dims: Tuple[Optional[int], Optional[int]],
+    ):
+        v2 = import_torchvision_transforms_v2()
+
+        assert isinstance(tv_random_crop, v2.RandomCrop)
+
+        if tv_random_crop.padding is not None:
+            raise ValueError(
+                "TorchVision RandomCrop transform must not specify padding."
+            )
+
+        if tv_random_crop.pad_if_needed is True:
+            raise ValueError(
+                "TorchVision RandomCrop transform must not specify pad_if_needed."
+            )
+
+        if tv_random_crop.fill != 0:
+            raise ValueError("TorchVision RandomCrop fill must be 0.")
+
+        if tv_random_crop.padding_mode != "constant":
+            raise ValueError("TorchVision RandomCrop padding_mode must be constant.")
+
+        if len(tv_random_crop.size) != 2:
+            raise ValueError(
+                "TorchVision RandcomCrop transform must have a (height, width) "
+                f"pair for the size, got {tv_random_crop.size}."
+            )
+
+        height, width = input_dims
+        if height is None:
+            raise ValueError(
+                "Video metadata has no height. RandomCrop can only be used when input frame dimensions are known."
+            )
+        if width is None:
+            raise ValueError(
+                "Video metadata has no width. RandomCrop can only be used when input frame dimensions are known."
+            )
+
+        # Note that TorchVision v2 transforms only accept NCHW tensors.
+        params = tv_random_crop.make_params(
+            torch.empty(size=(3, height, width), dtype=torch.uint8)
+        )
```

Comment:
That's OK for now, but creating a dummy input tensor is not ideal. we could consider exposing a private torchvision API that simply takes the input shape as parameter, rather than the input itself. Ultimately its shape is all that's needed.